File: webide/src/main/java/fiveguys/webide/api/user/dto/response/LoginUserResponse.java
package fiveguys.webide.api.user.dto.response;

import fiveguys.webide.config.auth.LoginUser;
import fiveguys.webide.domain.user.UserRole;
import lombok.Builder;
import lombok.Getter;

@Getter
public class LoginUserResponse {
    private Long userId;
    private String nickname;
    private UserRole role;
    private String accessToken;
    private String refreshToken;

    @Builder
    public LoginUserResponse(LoginUser loginUser, String accessToken, String refreshToken) {
        this.userId = loginUser.getUser().getId();
        this.nickname = loginUser.getUser().getNickname();
        this.role = loginUser.getUser().getRole();
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
    }
}

File: webide/src/main/kotlin/fiveguys/webide/user/dto/response/LoginUserResponse.kt
package fiveguys.webide.user.dto.response

import fiveguys.webide.config.auth.LoginUser
import fiveguys.webide.user.domain.UserRole

data class LoginUserResponse(
    val userId: Long?,
    val nickname: String,
    val role: UserRole,
    val accessToken: String,
    val refreshToken: String
) {
    constructor(loginUser: LoginUser, accessToken: String, refreshToken: String) : this(
        userId = loginUser.user.id,
        nickname = loginUser.user.nickname,
        role = loginUser.user.role,
        accessToken = accessToken,
        refreshToken = refreshToken
    )
}

